src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java

```diff
@@ -62,47 +76,50 @@ public String getMessageText() {
                 .getText();
     }
 
-    @Step("Speciality page: Deleted speciality by code")
-    public SpecialitiesPage deleteSpeciality(SpecialityEntity speciality) {
-        SelenideElement currentRow = findRowByColumnValue(speciality.getCode()); //consider code as unique field
-        currentRow.$x(".//i[contains(@class,'delete')]").click();
-        $x("//button/span[contains(text(),'Підтвердити')]").click();
+    @Step("Speciality page: Found table page with searched value")
+    public SpecialitiesPage findTablePageWithSearchValue(String searchValue) {
+        goToTablePage("first");
 
-        return this;
-    }
-
-    private SelenideElement findRowByColumnValue(String searchValue) {
-        SelenideElement currentRow;
-
-        if (buttonNavigationFirst.isEnabled()) {
-            buttonNavigationFirst.click();
-        }
-
-        currentRow = findRowOnCurrentPageByColumnValue(searchValue);
+        var buttonNavigationNext = $x(format(NAVIGATION_BUTTON_PATH_TEMPLATE, "next"));
+        boolean isSearchValueOnCurrentPage = false;
 
-        while (buttonNavigationNext.isEnabled() && currentRow == null) {
-            buttonNavigationNext.click();
+        while (buttonNavigationNext.isEnabled() && !isSearchValueOnCurrentPage) {
+            goToTablePage("next");
             buttonNavigationNext
                     .$x(".//div[contains(@class,'round')]/div[@class='mat-ripple-element']")
                     .should(disappear);
 
-            currentRow = findRowOnCurrentPageByColumnValue(searchValue);
+            isSearchValueOnCurrentPage = isSearchValueInTableTexts(searchValue);
         }
 
-        return currentRow;
+        return this;
+    }
+
+    @Step("Speciality page: Changed table page")
+    private void goToTablePage(String direction) {
```

Comment:
private methods should be placed after public ones

Reply:
It was supposed to be public( I'll fix